exporters/BxDRobotExporter/BxDRobotExporter/BxDRobotExporter/StandardAddInServer.cs

```diff
@@ -306,6 +309,38 @@ private void StartExporter()
             
             EnvironmentEnabled = true;
 
+
+            jointRelatedOccurences = new ArrayList();
+            foreach (ComponentOccurrence c in ((AssemblyDocument)MainApplication.ActiveDocument).ComponentDefinition.Occurrences)
```

Comment:
It's best practice to make object lists item and store the data as a copy before going into a iterative loop while doing casting and function calls.